optimum/utils/input_generators.py

```diff
@@ -858,3 +858,29 @@ def generate(self, input_name: str, framework: str = "pt", int_dtype: str = "int
             )
             for _ in range(self.num_layers)
         ]
+
+
+class FalconDummyPastKeyValuesGenerator(DummyPastKeyValuesGenerator):
+    def generate(self, input_name: str, framework: str = "pt", int_dtype: str = "int64", float_dtype: str = "fp32"):
+        self.num_kv_heads = 1
```

Comment:
Tiny falcon uses `self.new_decoder_architecture=True` and `self.multi_query=True` and in that case the number of kv heads is set to 1

```python
self.num_kv_heads = config.num_kv_heads if (self.new_decoder_architecture or not self.multi_query) else 1
```

https://github.com/huggingface/transformers/blob/main/src/transformers/models/falcon/modeling_falcon.py#L287C9-L287C114